functions/shared_libs/api_clients/eduhub_client.py

```
@@ -255,6 +258,115 @@ def fetch_enrollments(self, user_ids, course_id):
+    def _post_graphql(self, query, variables, operation_name):
+        """POSTs a GraphQL document with the admin secret and fails loudly.
+
+        Unlike `send_query`, this raises on a non-200 response and on a populated
+        `errors` key instead of returning a string / a half-empty payload. Use it
+        wherever silently returning no data would produce a wrong result rather
+        than a visible failure.
+
+        Returns:
+            dict: The `data` object of the response.
+        """
+        response = requests.post(
```

Ruff (0.16.1)

[error] Probable use of `requests` call without timeout

(S113)